chia/full_node/full_node_api.py

```diff
@@ -1273,13 +1273,32 @@ async def respond_compact_proof_of_time(self, request: timelord_protocol.Respond
     @execute_task
     @peer_required
     @api_request
-    async def new_compact_vdf(self, request: full_node_protocol.NewCompactVDF, peer: ws.WSChiaConnection):
+    @bytes_required
+    async def new_compact_vdf(
+        self, request: full_node_protocol.NewCompactVDF, peer: ws.WSChiaConnection, request_bytes: bytes = b""
+    ):
         if self.full_node.sync_store.get_sync_mode():
             return None
+
+        if len(self.full_node.compact_vdf_sem._waiters) > 20:
+            self.log.debug(f"Ignoring NewCompactVDF: {request}, _waiters")
+            return
+
+        name = std_hash(request_bytes)
+        if name in self.full_node.compact_vdf_requests:
+            self.log.debug(f"Ignoring NewCompactVDF: {request}, already requested")
+            return
+        self.full_node.compact_vdf_requests.add(name)
+
         # this semaphore will only allow a limited number of tasks call
         # new_compact_vdf() at a time, since it can be expensive
         async with self.full_node.compact_vdf_sem:
-            await self.full_node.new_compact_vdf(request, peer)
+            try:
+                await self.full_node.new_compact_vdf(request, peer)
+            except BaseException as e:
+                raise e
```

Comment:
What does catching and raising here achieve?  I see it adds the `raise e` line to the traceback, but maybe there's more?

```python
def f():
    raise Exception('from f()')

try:
    f()
except BaseException as e:
    raise e
```

```python-traceback
Traceback (most recent call last):
  File "main.py", line 7, in <module>
    raise e
  File "main.py", line 5, in <module>
    f()
  File "main.py", line 2, in f
    raise Exception('from f()')
Exception: from f()
```

Reply:
I added the try/catch/finally block not to catch exception per se, but to ensure that `name` get's removed from the set even if exception occurs. `self.full_node.compact_vdf_requests.remove(name)` 
I need to double check python syntax, catch can maybe be removed and just leave: 
```
try:
finally:
```

Reply:
`except` is optional, yes.  `try`/`finally` is fine.

Reply:
It isn't the better form in this case, but there is also just bare `raise` without the `e` which doesn't inject itself into the traceback.  Useful if you want to do something in response to the exception without actually handling it or interjecting more info into it.

Reply:
it seems to me that the `except ...` and `raise e` line aught to be removed.